Allow setting wildcard host in Apirule (#2620)
kyma-project/api-gateway: docs/user/expose-workloads/wildcard-host-name.md

## Requirements

- The referenced Gateway must serve a wildcard host that covers the domain you use in the APIRule.
- The `*` wildcard character is only allowed as the leftmost label of the host, for example, `*.example.com`. Patterns like `api.*.com` are
  not supported.
- Full regex host matching is not supported. If you need to match a specific pattern, use explicit hostnames in **spec.hosts** instead.

## Example

The following APIRule exposes a workload for any subdomain under `example.com`:

```yaml
apiVersion: gateway.kyma-project.io/v2
kind: APIRule
metadata:
  name: httpbin-wildcard
  namespace: default
spec:
  hosts:
    - "*.example.com”
  service:
    name: httpbin
    port: 8000
  gateway: kyma-system/kyma-gateway
  rules:
    - path: /headers
      methods: ["GET"]
      noAuth: true
```

Any request to a host matching `*.example.com` — for example `tenant1.example.com` or `tenant2.example.com` — is routed to the HTTPBin Service.

## Wildcard Host and Specific Host Conflict

> [!WARNING]

> > If a wildcard APIRule (for example, `*.example.com`) uses `noAuth`, it also matches specific subdomains like `a.example.com`. A separate APIRule for `a.example.com` with `jwt` will **not** enforce authentication, because the wildcard `noAuth` route already allows the request through.

Reviewer: ```suggestion
> If a wildcard APIRule (for example, `*.example.com`) uses `noAuth`, it also matches specific subdomains like `a.example.com`. In such a case, a separate APIRule for `a.example.com` with `jwt` does not enforce authentication, because the wildcard `noAuth` route already allows the request.
```

> To avoid this issue:

Reviewer: ```suggestion
To avoid the issue, follow these rules:
```
